### PR TITLE
docs(lsp): nits and typos in client.lua

### PR DESCRIPTION
**Pull request by MariaSolOs**
_Please don't judge me_.

### CODE REVIEW

File: runtime/lua/vim/lsp/client.lua [hunk 1]
```diff
@@ -113,10 +112,11 @@ local validate = vim.validate
...
 --- checking.
---- If {handler} is not specified,  If one is not found there, then an error
---- will occur. Returns: {status}, {[client_id]}. {status} is a boolean
---- indicating if the notification was successful. If it is `false`, then it
---- will always be `false` (the client has shutdown).
+--- If {handler} is not specified and if there's no respective server-side
+--- handler, then an error will occur.
+--- Returns: {status}, {[client_id]}. {status} is a boolean indicating if
+--- the notification was successful. If it is `false`, then it will always
+--- be `false` (the client has shutdown).
```
Comment (MariaSolOs) [inline runtime/lua/vim/lsp/client.lua:119]: This is the main change in this PR. @lewis6991 I found this description confusing, lmk if I got it right.

**Comment by MariaSolOs** [comment]
Pinging @lewis6991 ;)

File: runtime/lua/vim/lsp/client.lua (hunk 1 above)
Comment (lewis6991) [reply runtime/lua/vim/lsp/client.lua:207]: > if there's no respective server-side handler

This doesn't look correct to me?

The code is:
```lua
  if not handler then
    handler = assert(
      self:_resolve_handler(method),
      string.format('not found: %q request handler for client %q.', method, self.name)
    )
  end
```

and `_resolve_handler` is just:
```lua
function Client:_resolve_handler(method)
  return self.handlers[method] or lsp.handlers[method]
end
```

So I'd replace "server-side" handler with "global" or "default" handler.

**Comment by lewis6991** [comment]
Sorry I should have been more clear, I didn't mean literally "global or default", I meant "global" _or_ "default". Basically the handlers in `lsp/handlers.lua`. "default" maybe works better?

**Comment by MariaSolOs** [comment]
> Sorry I should have been more clear, I didn't mean literally "global or default", I meant "global" _or_ "default". Basically the handlers in `lsp/handlers.lua`. "default" maybe works better?

It's okay, my fault.

I'll use global since I see other places in the docs refer to the "global command registry".

**Pull request merged**